### PR TITLE
Support remote controlling Ykush devices

### PR DESCRIPTION
**Pull request by edersondisouza**
Currently, ykush devices can only be controlled 'locally'. This PR adds support to control them on remote devices. It is based on the seemingly abandoned https://github.com/labgrid-project/labgrid/pull/953 - I basically updated it with the comments there plus a minor fix - first patch, that could go on a different PR if deemed best.

Note that I also removed the 'Export' class created because I'm still figuring out its need - and the commentary for it. If it's deemed necessary, I can bring it back.

**Checklist**

- [X] The arguments and description in doc/configuration.rst have been updated
- [X] PR has been tested

### CODE REVIEW

**Comment by jluebbe** [comment]
> Note that I also removed the 'Export' class created because I'm still figuring out its need - and the commentary for it. If it's deemed necessary, I can bring it back.

The ResourceExport classes are needed to add the host and proxy information which makes local resources accessible via the network. That way, the exporter config can contain a `YKUSHPowerPort` entry, which is then converted to a `NetworkYKUSHPowerPort` by the exporter (which has the correct host and proxy information).

**Comment by edersondisouza** [comment]
v2:
- Re-add ResourceExport class for YKUSHPowerPower.

**Comment by edersondisouza** [comment]
Any comments about v2, @jluebbe?

File: labgrid/driver/powerdriver.py [hunk 1]
```diff
@@ -271,27 +271,37 @@ def get(self):
 @attr.s(eq=False)
 class YKUSHPowerDriver(Driver, PowerResetMixin, PowerProtocol):
     """YKUSHPowerDriver - Driver using a YEPKIT YKUSH switchable USB hub
-        to control a target's power - https://www.yepkit.com/products/ykush"""
+        to control a target's power - https://www.yepkit.com/products/ykush.
+        Uses ykushcmd tool to control the hub."""
     bindings = {"port": "YKUSHPowerPort", }
```
Comment (jluebbe) [inline labgrid/driver/powerdriver.py:276]: This needs to explicitly bind to `NetworkYKUSHPowerPort` as well.
```suggestion
    bindings = {"port": {"YKUSHPowerPort", "NetworkYKUSHPowerPort"} }
```
See for example the `FlashromDriver`.

File: labgrid/resource/ykushpowerport.py [hunk 2]
```diff
@@ -15,3 +15,8 @@ class YKUSHPowerPort(Resource):
     serial = attr.ib(validator=attr.validators.instance_of(str))
     index = attr.ib(validator=attr.validators.instance_of(int),
                     converter=int)
+
+@target_factory.reg_resource
+@attr.s(eq=False)
+class NetworkYKUSHPowerPort(NetworkResource, YKUSHPowerPort):
```
Comment (jluebbe) [inline labgrid/resource/ykushpowerport.py:21]: We don't use multiple inheritance for resources, so die users need explicitly choose which to support.

File: labgrid/remote/client.py [hunk 3]
```diff
@@ -744,6 +746,8 @@ def power(self):
                     drv = self._get_driver_or_new(target, "PDUDaemonDriver", name=name)
                 elif isinstance(resource, TasmotaPowerPort):
                     drv = self._get_driver_or_new(target, "TasmotaPowerDriver", name=name)
+                elif isinstance(resource, YKUSHPowerPort):
```
Comment (jluebbe) [inline labgrid/remote/client.py:749]:
```suggestion
                elif isinstance(resource, NetworkYKUSHPowerPort):
```

File: labgrid/remote/client.py [hunk 4]
```diff
@@ -727,13 +727,15 @@ def power(self):
         target = self._get_target(place)
         from ..resource.power import NetworkPowerPort, PDUDaemonPort
         from ..resource.remote import NetworkUSBPowerPort, NetworkSiSPMPowerPort
-        from ..resource import TasmotaPowerPort
+        from ..resource import TasmotaPowerPort, YKUSHPowerPort
```
Comment (jluebbe) [inline labgrid/remote/client.py:730]:
```suggestion
        from ..resource import TasmotaPowerPort, NetworkYKUSHPowerPort
```
Also add the import the the resource module.

**Review by jluebbe — CHANGES_REQUESTED** [review]
Please also add a hint to the changelog that it's now necessary to install the ykushcmd tool.

In `doc/configuration.rst`, there's still a recommendation to use `pykush -l` for the serial. Is there an alternative with ykushcmd?

File: labgrid/resource/ykushpowerport.py (hunk 2 above)
Comment (edersondisouza) [reply labgrid/resource/ykushpowerport.py]: Ack.

**Comment by edersondisouza** [comment]
v3:
  - No more inheritance from `YKUSHPowerPort` in `NetworkYKUSHPowerPort`;
  - Using `NetworkYKUSHPowerPort` directly on `client.py`;
  - `YKUSHPowerDriver` explicitly binds to `NetworkYKUSHPowerPort`;
  - Updated `configuration.rst` example to use `ykushcmd`;
  - Updated commit message to inform about installing `ykushcmd`.

**Comment by edersondisouza** [comment]
Any more coments, @jluebbe?

**Comment by edersondisouza** [comment]
Gentle reminder about this PR, @jluebbe =D

**Pull request merged**